addons/pos_self_order/controllers/orders.py

```diff
@@ -38,7 +38,11 @@ def process_order(self, order, access_token, table_identifier, device_type):
         order['fiscal_position_id'] = preset_id.fiscal_position_id.id if preset_id else pos_config.default_fiscal_position_id.id
         order['pricelist_id'] = preset_id.pricelist_id.id if preset_id else pos_config.pricelist_id.id
 
-        order = pos_config.env['pos.order'].sudo().with_company(pos_config.company_id.id).create([order])
+        existing_order = pos_config.env['pos.order'].search([('uuid', '=',order['uuid'])])
+        if not existing_order:
+            order = pos_config.env['pos.order'].sudo().with_company(pos_config.company_id.id).create([order])
+        else:
+            order = existing_order
```

Comment:
```suggestion
        order = pos_config.env['pos.order'].search([('uuid', '=',order['uuid'])]) || pos_config.env['pos.order'].sudo().with_company(pos_config.company_id.id).create([order])
```
this avoids the need for mutation